File: mobile/src/main/java/tr/xip/prayertimes/api/objects/Location.java
package tr.xip.prayertimes.api.objects;

import java.io.Serializable;

public class Location implements Serializable {
    int databaseId;
    String countryId;
    String countryName;
    String cityId;
    String cityName;
    String countyId;
    String countyName;

    int fajrNotification;
    int sunriseNotiication;
    int dhuhrNotification;
    int asrNotification;
    int maghribNotification;
    int ishaNotification;

    public Location(String countryId, String countryName, String cityId, String cityName, String countyId, String countyName) {
        this.countryId = countryId;
        this.countryName = countryName;
        this.cityId = cityId;
        this.cityName = cityName;
        this.countyId = countyId;
        this.countyName = countyName;
    }

    public Location(int databaseId, String countryId, String countryName, String cityId,
                    String cityName, String countyId, String countyName, int fajrNotification,
                    int sunriseNotiication, int dhuhrNotification, int asrNotification,
                    int maghribNotification, int ishaNotification) {
        this.databaseId = databaseId;
        this.countryId = countryId;
        this.countryName = countryName;
        this.cityId = cityId;
        this.cityName = cityName;
        this.countyId = countyId;
        this.countyName = countyName;
        this.fajrNotification = fajrNotification;
        this.sunriseNotiication = sunriseNotiication;
        this.dhuhrNotification = dhuhrNotification;
        this.asrNotification = asrNotification;
        this.maghribNotification = maghribNotification;
        this.ishaNotification = ishaNotification;
    }

    public int getDatabaseId() {
        return databaseId;
    }

    public String getCountryId() {
        return countryId;
    }

    public String getCountryName() {
        return countryName;
    }

    public String getCityId() {
        return cityId;
    }

    public String getCityName() {
        return cityName;
    }

    public String getCountyId() {
        return countyId;
    }

    public String getCountyName() {
        return countyName;
    }

    public String getName() {
        return countyName != null ? countyName : cityName;
    }

    public void setFajrNotification(int notification) {
        fajrNotification = notification;
    }

    public void setSunriseNotiication(int notification) {
        sunriseNotiication = notification;
    }

    public void setDhuhrNotification(int notification) {
        dhuhrNotification = notification;
    }

    public void setAsrNotification(int notification) {
        asrNotification = notification;
    }

    public void setMaghribNotification(int notification) {
        maghribNotification = notification;
    }

    public void setIshaNotification(int notification) {
        ishaNotification = notification;
    }

    public int getFajrNotification() {
        return fajrNotification;
    }

    public int getSunriseNotiication() {
        return sunriseNotiication;
    }

    public int getDhuhrNotification() {
        return dhuhrNotification;
    }

    public int getAsrNotification() {
        return asrNotification;
    }

    public int getMaghribNotification() {
        return maghribNotification;
    }

    public int getIshaNotification() {
        return ishaNotification;
    }
}

File: mobile/src/main/kotlin/tr/xip/prayertimes/model/Location.kt
package tr.xip.prayertimes.model

import java.io.Serializable

class Location : Serializable {
    var databaseId: Int = 0
        internal set
    var countryId: String?
    var countryName: String?
    var cityId: String?
    var cityName: String?
    var countyId: String?
    var countyName: String? = null

    var fajrNotification: Int? = 0
    var sunriseNotification: Int? = 0
    var dhuhrNotification: Int? = 0
    var asrNotification: Int? = 0
    var maghribNotification: Int? = 0
    var ishaNotification: Int? = 0

    val name: String?
        get() = countyName ?: cityName

    constructor(countryId: String?, countryName: String?, cityId: String?, cityName: String?, countyId: String?, countyName: String?) {
        this.countryId = countryId
        this.countryName = countryName
        this.cityId = cityId
        this.cityName = cityName
        this.countyId = countyId
        this.countyName = countyName
    }

    constructor(databaseId: Int?, countryId: String?, countryName: String?, cityId: String?,
                cityName: String?, countyId: String?, countyName: String?, fajrNotification: Int?,
                sunriseNotification: Int?, dhuhrNotification: Int?, asrNotification: Int?,
                maghribNotification: Int?, ishaNotification: Int?) {
        this.databaseId = databaseId ?: this.databaseId
        this.countryId = countryId
        this.countryName = countryName
        this.cityId = cityId
        this.cityName = cityName
        this.countyId = countyId
        this.countyName = countyName
        this.fajrNotification = fajrNotification
        this.sunriseNotification = sunriseNotification
        this.dhuhrNotification = dhuhrNotification
        this.asrNotification = asrNotification
        this.maghribNotification = maghribNotification
        this.ishaNotification = ishaNotification
    }
}
